rfcs/text/0008_pulse.md

```diff
@@ -0,0 +1,247 @@
+The mapping for those indices will be **`strict`** to avoid anyone storing unwanted/not-allowed info. The indexer defined in [the _Inject telemetry_ endpoint](#inject-telemetry) will need to handle accordingly the errors derived from the strict mapping.  
+We'll set up a process to add new mappings and their descriptions before every new release.
+
+#### Access control
+
+- The access to _raw_ data indices will be very limited. Only granted to those in need of troubleshooting the service and maintaining mappings (this is the Pulse/Telemetry team at the moment).
+- Special views (as in aggregations/visualisations/snapshots of the data stored in special indices via separated indexers/aggregators/ES transform or via _BigQuery_ or similar) will be defined for different roles in the company to help them to take informed decisions based on the data.  
+This way we'll be able to control "who can see what" on a very granual basis. It will also provide us with more flexibility to change to structure of the _raw_ if needed.
+
+### 2. Local Pulse Service
+
+This refers to the plugin running in Kibana in each of our customers' deployments. It will be a core plugin in NP, available for all others plugins to get the existing channels to send pieces of data and subscribe to instructions.
+
+The channel handlers are only defined inside the pulse context and are used to normalise the data for each channel before sending it to the remote service. The CODEOWNERS should notify the Pulse team every time there's an intended change in this context.
+
+#### Data storage
+
+For the purpose of transparency, we want the user to be able to retrieve the telemetry we send at any point, so we should store the information we send for each channel in their own local _dot_ internal indices (similar to a copy of the `pulse-raw-*` and `pulse-instructions-*` indices in our remote service). We may want to also sync back from the remote service any updates we do to the documents: enrichment of the document, anonymisation, categorisation when it makes sense in that specific channel, ...
+
+In the same effort, we could even provide some _dashboards_ in Kibana for specific roles in the cluster to understand more about their deployment.
+
+Only those specific roles (admin?) should have access to these local indices, unless they grant permissions to other users they want to share this information with.
+
+The users should be able to control how long they want to keep that information for (via ILM?).
+
+#### Sending telemetry
+
+The telemetry will be sent, preferably, from the server. Only falling back to the browser in case we detect the server is behind firewalls and it cannot reach the service or if the user explicitly sets the behaviour in the config.
+
+Periodically, the process (either in the server or the browser) will retrieve the telemetry to be sent by the channels, compile it into 1 bulk payload and send it encrypted to the [ingest endpoint](#inject-telemetry) explained earlier.
```

Comment:
A couple questions:
1. By "send it encrypted", do you mean application-level encryption before the data is sent via HTTPS? I think this is a must if the data is going to be routed through an end-user's browser.
2. The same goes for receiving instructions through an end-user's browser.
3. If an end-user's browser is going to be routing anything, we should limit that behavior to authenticated users only.

Reply:
1. Yes, the current telemetry applies some application-level encryption before sending it over HTTPS. 
2. Good point. I'll add a mention to it.
3. Right! I didn't mention it, but that's the way it is in telemetry nowadays. I'll make it clear in the docs.

Thank you! @jportner